### PR TITLE
Refactor database tests into individual test blocks

### LINKED ISSUE #634: Helpers to set up test connections
For local testing, we can lean on SQLite a lot more, and for macs at least, automatically look in some common brew paths to reduce the amount of setup needed.

### PR DESCRIPTION
**Pull request by hadley**
* Add `test_con()` helper and move individual database tests into own blocks
* Add `local_table()` helper

Fixes #634

### CODE REVIEW

File: tests/testthat/test-driver-oracle.R [hunk 1]
```diff
@@ -1,24 +1,20 @@
...
   # - Long/outstanding issue with batch inserting
   # date/datetime for Oracle.  See for example
   # #349, #350, #391
   # - There also looks like there are issues related
   # to binary elements of size zero.
   # - Finally, no boolean in Oracle prior to 23
   test_roundtrip(con, columns = c("time", "date", "datetime", "binary", "logical"))
+})
+
+test_that("can detect existance of table", {
```
Comment (simonpcouch) [inline tests/testthat/test-driver-oracle.R:12]:
```suggestion
test_that("can detect existence of table", {
```

File: tests/testthat/test-driver-sql-server.R [hunk 2]
```diff
@@ -79,271 +79,258 @@ test_that("SQLServer", {
...
-  local({
-    # Subseconds are retained upon insertion (#208)
-    con <- DBItest:::connect(DBItest:::get_default_context())
+test_that("Subseconds are retained upon insertion (#208)", {
+  con <- test_con("SQLSERVER")
+  data <- data.frame(time = Sys.time())
+  tbl <- local_table(con, "time", data)
 
-    data <- data.frame(time = Sys.time())
-    dbWriteTable(con, "time", data, field.types = list(time = "DATETIME"), overwrite = TRUE)
```
Comment (simonpcouch) [inline tests/testthat/test-driver-sql-server.R:141]: Is this `field.types = list(time = "DATETIME")` needed to retain the utility of the test?

File: tests/testthat/test-driver-sql-server.R [hunk 3]
```diff
@@ -79,271 +79,258 @@ test_that("SQLServer", {
...
+  expect_equal(values[-6], received[-6])
+  expect_equal(as.double(values[[6]]), as.double(received[[6]]))
+})
 
-  test_that("dates should always be interpreted in the system time zone (#398)", {
-    con <- DBItest:::connect(DBItest:::get_default_context(), timezone = "America/Chicago")
-    res <- dbGetQuery(con, "SELECT CAST(? AS date)", params = as.Date("2019-01-01"))
-    expect_equal(res[[1]], as.Date("2019-01-01"))
-  })
+test_that("can parse SQL server identifiers", {
```
Comment (simonpcouch) [inline tests/testthat/test-driver-sql-server.R:193]: this is pretty slick

**Review by simonpcouch — APPROVED** [review]
High-key, this PR is a W.

Love the new helpers.🕺

File: tests/testthat/test-driver-sql-server.R (hunk 2 above)
Comment (hadley) [reply tests/testthat/test-driver-sql-server.R:141]: I don't think so; POSIXct are already detected as date time

**Pull request merged**